dask/blockwise.py

```diff
@@ -14,9 +19,18 @@
     apply,
     stringify,
     stringify_collection_keys,
+    TaggedDispatch,
 )
 
 
+# Dispatch function for dynamic argument generation within
+# `Blockwise`.  All dispatch options must register by "tag",
+# and may only accept positional arguments (always ending with
+# the global index tuple).  For now, all arguments to the
+# registered function must be msgpack serializable.
+blockwise_io_dep_func = TaggedDispatch("generate_deps")
```

Comment:
@jrbourbeau - I am interested to get your thoughts on whether this is a "safe" approach for allowing the registration of multiple function for Blockwise+IO dependency generation at graph-materialization time.

Here, we introduce a new  `TaggedDispatch` class (similar to `Dispatch`, but the dispatching is now done by a "tag" rather than the type of the first argument).  For `Blockwise`,  we define a `blockwise_io_dep_func` instance of this class, and require the various Blockwise+IO layers to register a unique tag/funciton if it is possible to generate the dependencies at graph materialization time.

I am hoping this is safe, since the dispatch functions must be registered on the scheduler. Is my thinking correct?